keps/sig-node/3702-ima-namespace-support/README.md

```diff
@@ -0,0 +1,263 @@
+As a cluster admin, I want to detect undesired file changes, so that I can take out pods that have been compromised.
+
+An intruder perform some malicious changes inside a certain pod's files. The system should be able to detect those changed and alert about the inconsistent pod. The remote attestation framework could keep this pod running or make a copy (for forensic analysis) and delete it.
+
+#### Story 2
+As a cluster admin, I want to deploy a only proven and certified pods, so that I can comply with internal policies as well as security regulations.
+
+Let's say that we are working in a high security environment where only approved images can be deployed. In this scenario we can make sure that the pod deploy used an imaged that hasn't been tampered.
+
+#### Story 3
+As a cluster admin, I want to deny access to certain files inside the pod, so that a potential intruder can't access sensitive information.
+
+In some cases, we should not even allow root to modify certain files inside the container.
+
+### Notes/Constraints/Caveats (Optional)
+We need to enable IMA in the kernel and container runtimes (runC, CRI-O, docker, containerd, etc.). 
+
+### Risks and Mitigations
+
+  
+
+## Design Details
+  
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+In order for the this feature to work, the nodes where the pods will be deploy should have IMA enabled and a recent kernel that supports IMA namespaces (WIP, it should be merged soon). The nodes should also have a TPM chip. We could use nodAffinity of labels and annotations in nodes in order to select where to deploy the pods.
+
+The linux kernel IMA namespace support is based on user namespaces. Therefore, the container runtime should first create a user namespace and then create an IMA namespaces. In order to use IMA namespaces it is necessary to enable user namespaces as well.
+
+Should we enable IMA namespaces by default when enabling user namespaces?
+
+
+There will be a CRI API change which will allow the pod to use IMA namespaces and specify the namespace policy.
+
+  
+
+### Linux kernel
+
+  
+
+IMA is only available in Linux hosts and Linux containers. Unfortunately, IMA is not a separate namespace, which is needed in order to isolate it and be used inside containers. Upcoming kernel patches should add support for IMA namespaces.
+
+  
+
+### Runtime specification
+
+There is an ongoing discussion regarding the runtime changes.
+
+https://github.com/opencontainers/runc/pull/3639
+ 
+
+### CRI API
+
+  
+
+We propose to add the following message.
+
+  
+
+```protobuf
+
+message NamespaceOptions {
+  bool  ima = 6;
+}
+
+```
+### Kubernetes pod resource
+We propose the following change in the podTemplate to enable IMA namespaces.
```

Comment:
pod.spec.

I think this and some others I've already mentioned in my previous review.